File: frontend/src/api/mapRoute.ts
// frontend/src/api/mapRoute.ts
// API client for trip map visualization data.

import api from "@/api/axiosInstance";

export interface StopMapPoint {
  stop_id: string;
  position_order: number;
  city_name: string;
  country: string;
  latitude: number;
  longitude: number;
  start_date: string;
  end_date: string;
  nights: number;
  is_overnight: boolean;
  activities_count: number;
  estimated_daily_cost: number;
  accommodation_suggestion: string;
  best_area_to_stay: string;
}

export interface SegmentInfo {
  from_city: string;
  to_city: string;
  from_coords: [number, number];
  to_coords: [number, number];
  distance_km: number;
  suggested_transport: string;
  estimated_travel_hours: number;
}

export interface TripMapData {
  trip_id: string;
  trip_name: string;
  total_days: number;
  total_distance_km: number;
  stops: StopMapPoint[];
  segments: SegmentInfo[];
  map_center: [number, number];
  map_zoom: number;
}

export const mapRouteApi = {
  getMapData: (tripId: string) =>
    api.get<TripMapData>(`/trips/${tripId}/map-data`),
};

File: backend/app/schemas/map_route.py
# backend/app/schemas/map_route.py
# Pydantic schemas for trip map/route visualization data.

from datetime import date
from pydantic import BaseModel


class StopMapPoint(BaseModel):
    stop_id: str
    position_order: int
    city_name: str
    country: str
    latitude: float
    longitude: float
    start_date: date
    end_date: date
    nights: int
    is_overnight: bool
    activities_count: int
    estimated_daily_cost: float
    accommodation_suggestion: str
    best_area_to_stay: str


class SegmentInfo(BaseModel):
    from_city: str
    to_city: str
    from_coords: list[float]
    to_coords: list[float]
    distance_km: float
    suggested_transport: str
    estimated_travel_hours: float


class TripMapData(BaseModel):
    trip_id: str
    trip_name: str
    total_days: int
    total_distance_km: float
    stops: list[StopMapPoint]
    segments: list[SegmentInfo]
    map_center: list[float]
    map_zoom: int
